devtools/projects/ng-devtools/src/lib/devtools-tabs/profiler/timeline/record-formatter/bargraph-formatter/bargraph-formatter.ts

```diff
@@ -16,22 +16,48 @@ export interface BargraphNode {
   value: number;
   label: string;
   original: ElementProfile;
+
+  directives?: DirectiveProfile[];
+  count?: number;  // number of merged nodes with the same label
 }
 
 export class BarGraphFormatter extends RecordFormatter<BargraphNode[]> {
   @memo({cache: new WeakMap()})
   override formatFrame(frame: ProfilerFrame): BargraphNode[] {
     const result: BargraphNode[] = [];
     this.addFrame(result, frame.directives);
-    return result.filter((element) => element.value > 0).sort((a, b) => b.value - a.value);
+    // Remove nodes with 0 value.
+    const nodesWithValue = result.filter((element) => element.value > 0);
+
+    // Merge nodes with the same label.
+    const labelToBarGraphNodes: {[key: string]: BargraphNode} = {};
+    nodesWithValue.forEach((node) => {
+      if (labelToBarGraphNodes[node.label] === undefined) {
+        labelToBarGraphNodes[node.label] = {
+          label: node.label,
+          value: node.value,
+          original: node.original,
+          parents: [],
+          directives: node.original.directives || [],
```

Comment:
nit: let's use `??` instead. 
```suggestion
          directives: node.original.directives ?? [],
```